src/ui/public/directives/file_upload.js

```diff
@@ -1,32 +1,80 @@
+import _ from 'lodash';
 import $ from 'jquery';
 import uiModules from 'ui/modules';
 var module = uiModules.get('kibana');
 
-module.directive('fileUpload', function ($parse) {
+var html = '<span class="dropzone" ng-transclude></span>';
+
+module.directive('fileUpload', function () {
   return {
-    restrict: 'A',
+    restrict: 'E',
+    transclude: true,
+    scope: {
+      onRead: '&',
+      onLocate: '&',
+      label: '='
+    },
+    template: html,
     link: function ($scope, $elem, attrs) {
-      var onUpload = $parse(attrs.fileUpload);
+      var $button = $elem.find('.upload');
```

Comment:
Hmm, seems to me like this is a slightly odd API... So essentially if I include something in the directive that has a class of "upload" it will be turned into something clickable with a file input. I'm not sure what the best alternative is, though, because maybe the consumer of this directive would want a link, or a button, so you can't just look for all links or all buttons either... I just feel like this should be part of the API somehow (like one of the `scope` values, or something). I don't know for sure, I just think this seems a little odd. I'll think on it some more.